### ISSUE TITLE
Support intra-doc linking to `()`

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust#144937]
# Code

```rust
/// [`()`] is the best type ever!
```

# Expected Outcome
A link to the unit type.

# Actual Output
Dead link appearing as "[`()`]".

# Version
rustdoc 1.89.0 (29483883e 2025-08-04)
rustdoc 1.91.0-nightly (0060d5a2a 2025-08-04)

# Additional Details

The link can be written as ``[`()`][primitive@unit]`` but I think this should not be necessary, since `()` is an unambiguous, non-generic type name.

@rustbot label +A-intra-doc-links

**Comment by imladenov51** [comment]
Looks like someone already was on this and said there were too many false positives for `()`. See #80181 for more information.
